### ISSUE TITLE
Unclear readme and only 4.20 version

### ISSUE THREAD

**Issue by jojothebandit** [JetBrains/RiderSourceCodeAccess#1]
Hello!
I would like to try Rider out with UE but I didn't manage to get it set up.
I find this piece of doc quite confusing as to what exactly I should copy where:
```
Copy RiderSourceCodeAccess folder:
    Engine from sources: to {UnrealEngineRoot}/Engine/Plugins/Developer or to {UnrealProjectRoot}/Plugins/Developer;
    Engine from Epic store: to {UnrealProjectRoot}/Plugins/Developer;
```

I fiddled around with it as I'm familiar with plugin development, and managed to get it "installed" as in it appears in the plugins list as enabled in UE4

Another thing that might be a problem is that I get a message 

"The RiderSourceCodeAccess plugin was designed for build 4.20, attempt to load it anyway?"
I said yes, i said no, i tried manually editing the uplugin file but didn't manage to get it functional.
I am using 4.24.3 UE on Windows 
Any ideas as to how I could try it out :)?

Thank you!

**Comment by ericwomer** [comment]
There is a ue_4_24 branch.

**Comment by SuperWig** [comment]
>I am using 4.24.3 UE on Windows

Use the releases version
https://github.com/JetBrains/RiderSourceCodeAccess/releases

**Comment by LukeSolas** [comment]
Well i have the same problem and I am using the most recent release version...
When trying to start up UE4 after adding the RiderSourceCodeAccess Folders in the Engine Root as well as the Project Root I get the following Error:

Plugin 'RiderSourceCodeAcess' failed to load because module 'RiderSourceCodeAcess' could not be found. 

When trying to upen up the uproject itself it tells me the module is missing... have I overlooked something?

**Comment by ericwomer** [comment]
@LukeSolas Your missing the instructions, from README.md

How to add to Game project:

1. Go to "Releases" page and download version of plugin for your version of Unreal Engine
2. Unzip RiderSourceCodeAccess.zip to RiderSourceCodeAccess folder
3. Copy RiderSourceCodeAccess folder to {UnrealProjectRoot}/Plugins/Developer; a. If {UnrealProjectRoot}/Plugins/Developer does not exist, create it
4. Re-generate solution files - uproject file > context menu > Generate Visual Studio project files
5. Build UE project using Rider or Visual Studio
6. Start Unreal Editor
7. Edit > Editor Preferences ... > General > Source Code > Source Code Editor
8. Select Rider from drop down list. NB: only Rider with C++ plugins will be available from drop down list

**Comment by LukeSolas** [comment]
Hm no I think I did that. The problem might be step 5.  I'll take a look at that later.
Thanks for the help anyway.

**Comment by ericwomer** [comment]
> Hm no I think I did that. The problem might be step 5. I'll take a look at that later.
> Thanks for the help anyway.

You said you put the RiderSourceCodeAccess in your project or engine root folders and it didn't work, you need to place it in either your {UnrealGameProjectRoot}/Plugins/Developer folder, or {UnrealEngineRoot}/Engine/Plugins/Developer, and load up the editor and in the editor settings set your source code accessor to Rider.

**Comment by LukeSolas** [comment]
Well still doesn't work. Still RiderSourceCodeAccess is missing or built in a different engine version.
Don't really know how I do step 5, simply make a console project in VS or Rider?
Admittedly, Im just starting out with working in IDE's and Engines, so bear with me :)
Thanks for the help in any case.

**Comment by ericwomer** [comment]
Step five is after you created your project, the rest of it is spelled out plainly in the readme, you have to download the correct version for your editor and you have to install it into the correct place, not your projects rood dir, for the editor to find it.

**Comment by tonymrakovcic** [comment]
Hi, thanks for the help, there seems to be a crucial part we're missing.
I see the plugin as enabled in UE but it doesn't show up on my options for an editor
I put everything as mentioned. Tried both the engine root and the projectroot/plugins/developer, regenerated with vs and it doesnt show up in the list

**Comment by LukeSolas** [comment]
The Problem solved itself. Rider ist now available in the Epic Marketplace
as a Plugin.
Install that and IT will Work


Tony Mrakovcic <notifications@github.com> schrieb am Mi., 29. Apr. 2020,
20:08:

**Comment by tonymrakovcic** [comment]
unfortunately no cigar :(
Here are some screenshots, I installed it through the marketplace and still no rider in the list
Do you have any idea what it could be?
Windows 10, UE 4.24.3
![riderinstalled](https://user-images.githubusercontent.com/3583441/80693500-62e30880-8ad3-11ea-8e2d-bc572e7f7f80.png)
![ridernotinlist](https://user-images.githubusercontent.com/3583441/80693501-637b9f00-8ad3-11ea-8e7a-5913dcb169bd.png)

**Comment by DecoyRS** [comment]
@tonymrakovcic do you have "Rider for Unreal Engine" installed? Regular "Rider" won't show up because it doesn't have Unreal Engine and/or C++ support

**Comment by tonymrakovcic** [comment]
Installed it now, didnt know i need to have that exact version. Was happy thinking it would work now. Nope :(
Tried reinstalling plugin, regenerated files, built with VS, recompiled module, tried every combination. I cant get rider into that list :D

**Comment by DecoyRS** [comment]
@tonymrakovcic , let's check it step by step:
* do you have "Rider for Unreal Engine" installed, provided from this link: https://www.jetbrains.com/rider/unreal/ ?
* what version of UE do you have? I assume 4.24 from the screenshot
* do you use pre-built Engine from EGS or did you built it from source from GitHub?
* have you deleted all other versionf of RiderSourceCodeAccess before installing from marketplace?

**Comment by tonymrakovcic** [comment]
**Do you have "Rider for Unreal Engine" installed, provided from this link: https://www.jetbrains.com/rider/unreal/ ?**
Yes, just freshly installed it

**what version of UE do you have? I assume 4.24 from the screenshot**

Yes, 4.24.3
**do you use pre-built Engine from EGS or did you built it from source from GitHub?**
Prebuilt

**have you deleted all other version of RiderSourceCodeAccess before installing from marketplace?**
I think i have. I dont have an RiderSourceCodeAccess in my engine folder
I've tried the same project, a completely new project, recompiling a few times, both the project and the module
Maybe tell me exactly where to look so I can try it out

**Comment by ericwomer** [comment]
> **Do you have "Rider for Unreal Engine" installed, provided from this link: https://www.jetbrains.com/rider/unreal/ ?**
> I think i have. I dont have an RiderSourceCodeAccess in my engine folder
> I've tried the same project, a completely new project, recompiling a few times, both the project and the module

Its not supposed to be in your Engine folder, its supposed to be in {UnrealProjectRoot}/Plugins/Developer (or on with source built engine {UnrealProjectRoot}/Engine/Plugins/Developer)

> Maybe tell me exactly where to look so I can try it out

{UnrealProjectRoot}/Plugins/Developer or {UnrealProjectRoot}/Engine/Plugins/Developer

**Comment by tonymrakovcic** [comment]
Thats what I was doing when it wasnt on the marketplace.
Ive tried this now again (so installed as marketplace plugin, AND added the zip contents to {UnrealProjectRoot}/Engine/Plugins/Developer  ), but as always, plugin installed, plugin enabled - but no rider in the editor list

**Comment by DecoyRS** [comment]
@tonymrakovcic can you open Rider, go to  Help > About and share a screenshot of your popup?

**Comment by tonymrakovcic** [comment]
I guess you mean riderunreal,  i have rider 2020.1.1 installed from the toolbox also

![riderihelp](https://user-images.githubusercontent.com/3583441/80725540-802fcb00-8b03-11ea-9f91-de6648c8c007.png)

**Comment by tonymrakovcic** [comment]
I went so far as to looking into the .cpp files and try stuff out
Withh admin permissions, without admin permissions, fiddled with registry paths, installed java (since there's a jar file in there somewhere), added symlink (since it can be searching in the wrong dir for the exe) and simply cannot get "Rider" as part of the list in the source code editor dropdown :D
I think I am out of options, if you have something I can test out if it'll work, I'll be happy to, since I would really like to try it out

**Comment by DecoyRS** [comment]
@tonymrakovcic if you don't mind, we can have a video call and go through step by step options to check what's not working for you) dm me here (if it's even possible) or on Tweeter (Pie_Daddy)

**Comment by tonymrakovcic** [comment]
Sure, I've added you on twitter, I'm jojothebandit, a video call would be okay tomorrow, but if you can give me some options to check out before, just to clear out the obvious stuff.

**Comment by tonymrakovcic** [comment]
We managed to get this fixed!
It's still a little mysterious, but it seems that Rider must be installed with admin privileges for this to work out of the box. The first time I installed it it was not an admin account, and thats the only thing I didnt try, reinstalling riderUE with admin privileges. Alhotugh I'm still not 100% this is the case, it seems to be working now.
Thank you all for your help!
